tidy3d/components/field_projection.py

```diff
@@ -534,6 +559,10 @@ def project_fields(
             return self._project_fields_cartesian(proj_monitor)
         return self._project_fields_kspace(proj_monitor)
 
+    def is_3d_simulation(self) -> bool:
+        """Determine if the simulation is 2D or 3D based on the size attribute."""
+        return self.sim_data.simulation.size[2] != 0.01
```

Comment:
I assume this was set for your specific test? Normally we check if `sim.size[index] > 0`. We could also consider `if sim.grid.num_cells[index] > 1` to capture more general cases with a single pixel along a direction.

Reply:
Thank you for the comments, @momchil-flex ! To achieve that goal for a general case, I'm considering how to modify the monitor if we allow 0-dim for every dimension. For example, Ez/Hphi and Hz/Ephi are generated when z is 0-dim, Ey/Hphi and Hy/Ephi are generated when y is 0-dim, and the same applies for x being 0-dim. Therefore, we would need to introduce new fields in the field monitor to consider all these three cases. Is this correct?

Reply:
The condition `if sim.grid.num_cells[index] > 1` seems to fit well!